### PR TITLE
rbd-mirror: include local pool id in resync throttle unique key

### PR DESCRIPTION
**Pull request by dillaman**
Fixes: http://tracker.ceph.com/issues/16536
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

**Comment by rjfd** [comment]
lgtm

File: src/tools/rbd_mirror/ImageSyncThrottler.cc [hunk 1]
```diff
@@ -88,33 +90,35 @@ void ImageSyncThrottler<I>::start_sync(
...
 
   if (start) {
-      sync_holder_ctx->m_sync->send();
+    sync_holder_ctx->m_sync->send();
   }
 }
 
 template <typename I>
-void ImageSyncThrottler<I>::cancel_sync(const std::string& local_image_id) {
+void ImageSyncThrottler<I>::cancel_sync(librados::IoCtx &local_io_ctx,
```
Comment (rjfd) [inline src/tools/rbd_mirror/ImageSyncThrottler.cc]: Why not just pass the uint64_t pool id instead of the IoCtx object?

Comment (dillaman) [inline src/tools/rbd_mirror/ImageSyncThrottler.cc]: Just to abstract away how the key is formed -- I originally wanted to change the signature to pass in ImageCtxT but that would have involved more work in bootstrap to only cancel if an image sync was scheduled.

**Comment by trociny** [comment]
lgtm

**Pull request merged**